packages/spacecat-shared-data-access/src/models/importer/import-url.js

```diff
@@ -21,78 +21,72 @@ export const ImportUrlStatus = {
 };
 
 /**
- * Creates a new ImportUrl object
- *
- * @param {Object} importUrlData
- * @returns {ImportUrl}
+ * Creates a new ImportUrl object.
+ * @param {Object} data - The data for the ImportUrl object.
+ * @returns {Readonly<{ImportUrl}>} The new ImportUrl object.
  */
 const ImportUrl = (data) => {
-  const self = Base(data);
+  const importUrl = Base(data);
 
-  self.getJobId = () => self.state.jobId;
-  self.getUrl = () => self.state.url;
-  self.getStatus = () => self.state.status;
-  self.getReason = () => self.state.reason;
+  importUrl.getJobId = () => importUrl.state.jobId;
+  importUrl.getUrl = () => importUrl.state.url;
+  importUrl.getStatus = () => importUrl.state.status;
+  importUrl.getReason = () => importUrl.state.reason;
   // Absolute path to the resource that is being imported for the given URL
-  self.getPath = () => self.state.path;
+  importUrl.getPath = () => importUrl.state.path;
   // Resulting path and filename of the imported .docx file
-  self.getFile = () => self.state.file;
+  importUrl.getFile = () => importUrl.state.file;
 
   /**
-   * Updates the status of the ImportUrl
+   * Updates the state of the ImportJob.
+   * @param key - The key to update.
+   * @param value - The new value.
+   * @param validator - An optional validation function to use before updating the value.
+   * The validator can return false to indicate that the value isn't worth throwing an exception,
+   * but continue to use the previous value.
    */
-  self.setStatus = (status) => {
-    if (!Object.values(ImportUrlStatus).includes(status)) {
-      throw new Error(`Invalid Import URL status during update: ${status}`);
+  const updateState = (key, value, validator) => {
+    if (validator && typeof validator === 'function') {
+      // a validator can return true or false to indicate if the value is valid
+      // however if a validator throws an error, it is considered critical and invalid.
+      if (!validator(value)) {
```

Comment:
Do we need to consider a similar check for import-job?

Reply:
It doesn't seem to be the same use case from what I saw.  In the ImportUrl scenario the use of hasText, was used to validate the input, but does not throw an error.  In the case of ImportJob, any missing or invalid data did throw an error, so the use case to continue if the value was false wasn't there.  So I kept the existing behaviour not to shake the tree to much.